app/src/main/java/com/medmapper/v33001/dto/Medicine.kt

```diff
@@ -6,10 +6,12 @@ import androidx.room.PrimaryKey
 
 @Entity(tableName = "Medicine")
 data class Medicine(
-    @PrimaryKey(autoGenerate = true) val medID: Int,
-    @ColumnInfo(name = "name") val name: String?,
+    @PrimaryKey(autoGenerate = true) var medID: Int,
```

Comment:
My mistake, confused mutable var vs val.

Reply:
Keep this as val.  That's the default.  Only make var if it will be reassigned.